docs/src/main/asciidoc/application-configuration-guide.adoc

```diff
@@ -360,6 +360,45 @@ You can override these runtime properties with the following mechanisms (in decr
 
 NOTE: Environment variables names are following the conversion rules of link:https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configsources.asciidoc#default-configsources[Eclipse MicroProfile]
 
+== Using DotEnv files during Development
+As an alternative to defining global environment variables, you can define environment variables in a `.env` file.
+This is practical as it allows each member of your development team to customize and manage the development environment via a file.
+Let's see this in action.
```

Comment:
I think that we need to be clear here that this doesn't really define "real" process environment variables (which are immutable in Java).  Instead of using the phrase "environment variables", we should use the phrase "local configuration properties" or something like that.  In fact there is no relationship between environment variables and this implementation of `.env` other than the fact that both become config sources.

Reply:
Thanks @dmlloyd this is a good point. I'll change the wording.

Reply:
shouldn't we call these `.properties` or similar instead ? 
`.env` are used by other tools to setup environment variables, i.e. docker-compose.

Reply:
never mind - misread the PR - these aren't properties; they are actually env variables rather than overriding application.properties

Reply:
You were right the first time.  Really this is just a config source in `.properties` format, it's just called `.env` for convention.  But if your process calls `System.getenv()`, none of these values will show up.

Reply:
@dmlloyd would you say we change just the wording in the documentation, and the `-Dquarkus.env.file` know to something like `-Dquarkus.local-config` ?

Reply:
Possibly.  I wonder if @maxandersen's concern about overlap with docker-compose and similar tools would cause an actual problem for users.

Having a base name of `application.properties` is a bit awkward in this regard because there's no real logical shortening of this.

Maybe the file ought to be called e.g. `.quarkus` or something instead, and the local config property ought to be called `quarkus.local-config` as you suggest, or maybe `quarkus.config.local`.  Or if we want to try to match up with the original name, `.application` or even `.application.properties`?  I don't know.

Reply:
> Possibly. I wonder if @maxandersen's concern about overlap with docker-compose and similar tools would cause an actual problem for users.

Won't these configuration be part of host/container environment variables? In this case, they will take precedence over .env properties. 

> Maybe the file ought to be called e.g. `.quarkus` or something instead, and the local config property ought to be called `quarkus.local-config` as you suggest, or maybe `quarkus.config.local`. Or if we want to try to match up with the original name, `.application` or even `.application.properties`? I don't know.

I like the `.application.properties` suggestion, I'll go with this one. 
Also users may call the file however they see fit as long as it is a valid `.properties` file when using the `quarkus.local-config` config property.

Reply:
> > Possibly. I wonder if @maxandersen's concern about overlap with docker-compose and similar tools would cause an actual problem for users.
> 
> Won't these configuration be part of host/container environment variables? In this case, they will take precedence over .env properties.

Sure, if that's how you code it. :slightly_smiling_face:

> > Maybe the file ought to be called e.g. `.quarkus` or something instead, and the local config property ought to be called `quarkus.local-config` as you suggest, or maybe `quarkus.config.local`. Or if we want to try to match up with the original name, `.application` or even `.application.properties`? I don't know.
> 
> I like the `.application.properties` suggestion, I'll go with this one.
> Also users may call the file however they see fit as long as it is a valid `.properties` file when using the `quarkus.local-config` config property.

My only comment then is that the priority should not be equal to the priority of any other pre-defined configuration source.  You'll want a little bit of space in between so that users can do whatever crazy hacks they want to do.  Maybe we should have an overall configuration source priority list somewhere...

Reply:
> My only comment then is that the priority should not be equal to the priority of any other pre-defined configuration source. You'll want a little bit of space in between so that users can do whatever crazy hacks they want to do. Maybe we should have an overall configuration source priority list somewhere...

I see your point, I chose an arbitrary value of `299`, I can modify that too to `290` maybe :-)

Reply:
Our ordering is constrained at least in part by the MP config specification.  But I think it'd be good to get it all documented in one place regardless.

Reply:
@dmlloyd I have force pushed new changes. WDYT?